skills/token/base.py

```diff
@@ -36,25 +36,38 @@ def get_api_key(self, context: SkillContext) -> str:
 
         Returns:
             The API key to use for API requests
+
+        Raises:
+            ValueError: If no API key is available
         """
         skill_config = context.config
         if skill_config.get("api_key_provider") == "agent_owner":
-            return skill_config.get("api_key")
-        return self.skill_store.get_system_config("moralis_api_key")
+            api_key = skill_config.get("api_key")
+            if not api_key:
+                raise ValueError(
+                    "No agent-specific API key provided in the configuration."
+                )
+            return api_key
+
+        api_key = self.skill_store.get_system_config("moralis_api_key")
+        if not api_key:
+            raise ValueError("No Moralis API key provided in the configuration.")
+        return api_key
 
     def context_from_config(self, config: Optional[RunnableConfig] = None) -> Any:
```

Comment:
Unless necessary, do not override common functions. This way, if there are any updates upstream, they will not be overwritten here.